File: php/listar-dominios.php
<?php

// Definição do endpoint da API para listar domínios
$url = "https://api.mailgrid.net.br/domain/list/";

// Dados de autenticação (substitua pelo seu token real)
$data = [
    "token_auth" => "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" // Token obtido no painel do cliente
];

// Configuração da requisição cURL
$options = [
    CURLOPT_URL => $url, // Define a URL do endpoint da API
    CURLOPT_RETURNTRANSFER => true, // Retorna a resposta como string
    CURLOPT_POST => true, // Define o método da requisição como POST
    CURLOPT_HTTPHEADER => ["Content-Type: application/json"], // Define o cabeçalho da requisição como JSON
    CURLOPT_POSTFIELDS => json_encode($data), // Converte os dados para JSON antes de enviar
    CURLOPT_TIMEOUT => 30, // Tempo limite da requisição em segundos
    CURLOPT_SSL_VERIFYPEER => true // Mantém a verificação SSL ativada para segurança
];

// Inicializa o cURL
$ch = curl_init();
curl_setopt_array($ch, $options);
$response = curl_exec($ch); // Executa a requisição e captura a resposta
$statusCode = curl_getinfo($ch, CURLINFO_HTTP_CODE); // Obtém o código de status HTTP da resposta

// Verifica se houve erro na requisição
if (curl_errno($ch)) {
    echo "Erro na requisição: " . curl_error($ch); // Exibe a mensagem de erro do cURL
} elseif ($statusCode >= 400) {
    echo "Erro na requisição. Código de status HTTP: " . $statusCode; // Exibe o código de erro HTTP
} else {
    // Decodifica a resposta JSON para um array associativo
    $decodedResponse = json_decode($response, true);
    
    // Verifica se a resposta foi decodificada corretamente
    if (json_last_error() === JSON_ERROR_NONE) {
        echo "Resposta da API:\n";
        echo json_encode($decodedResponse, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE); // Exibe a resposta formatada
    } else {
        echo "Erro ao processar a resposta JSON."; // Mensagem de erro caso a resposta JSON seja inválida
    }
}

// Fecha a conexão cURL para liberar recursos
curl_close($ch);

?>

File: python/listar-dominios.py

import requests
import json

# Definição do endpoint da API para listar domínios
url = "https://api.mailgrid.net.br/domain/list/"

# Dados de autenticação (substitua pelo seu token real)
data = {
	"token_auth": "SEU_TOKEN_AQUI"  # Token obtido no painel do cliente
}

# Cabeçalhos para a requisição (informando que estamos enviando dados em JSON)
headers = {
	"Content-Type": "application/json"
}

# Fazendo a requisição POST para a API
try:
	# Realiza a requisição POST, enviando os dados no formato JSON
	response = requests.post(url, headers=headers, json=data, timeout=30)
	
	# Verifica o código de status da resposta
	if response.status_code >= 400:
		print(f"Erro na requisição. Código de status HTTP: {response.status_code}")
	else:
		# Decodifica e exibe a resposta formatada em JSON
		try:
			decoded_response = response.json()  # Decodifica a resposta JSON
			print("Resposta da API:")
			# Exibe a resposta formatada
			print(json.dumps(decoded_response, indent=4, ensure_ascii=False))
		except json.JSONDecodeError:
			print("Erro ao processar a resposta JSON.")
except requests.exceptions.RequestException as e:
	# Captura qualquer erro relacionado à requisição (erros de rede, tempo limite, etc)
	print(f"Erro na requisição: {e}")
		
